src/runtime/vm/executable.cc

```diff
@@ -476,8 +484,19 @@ void LoadHeader(dmlc::Stream* strm) {
 }
 
 runtime::Module Executable::Load(const std::string& code, const runtime::Module lib) {
+  std::cout << "code: " << code.size() << std::endl;
   auto exec = make_object<Executable>();
-  exec->lib = lib;
+
+  // Support null-initialization of lib, to enable initialization during
+  // deserialization before we have we have deserialized the imports.
+  if (lib.defined()) {
+    ICHECK_EQ(exec->imports_.size(), 0)
```

Comment:
maybe this should move to GetLib. Since you just made exec, it should not have imports?

Reply:
It shouldn't but I'm a defensive programmer and paranoid about people wrongfactoring the code later in a way that breaks an invariant.